src/circuit.jl

```diff
@@ -271,6 +271,7 @@ QubitSet with 2 elements:
 qubits(c::Circuit) = (qs = union!(copy(c.moments._qubits), c.qubit_observable_set); QubitSet(qs))
 function qubits(p::Program)
     inst_qubits = mapreduce(ix->ix.target, union, p.instructions, init=Set{Int}())
+    bri_qubits  = mapreduce(ix->ix.target, union, p.basis_rotation_instructions, init=Set{Int}())
     res_qubits  = mapreduce(ix->(hasproperty(ix, :targets) && !isnothing(ix.targets)) ? reduce(vcat, ix.targets) : Set{Int}(), union, p.results, init=Set{Int}())
     return union(inst_qubits, res_qubits)
```

Comment:
do you need to also return `bri_qubits`?

Reply:
Good catch, yep

Reply:
ok, I'll approve this, up to you whether you want to fix this in this PR or a follow-up